cargo-flash/src/main.rs

```diff
@@ -104,10 +109,15 @@ fn main_try() -> Result<(), failure::Error> {
         None => panic!(),
     };
 
-    let mut args = std::env::args();
+    let mut args: Vec<_> = std::env::args().collect();
     // Remove first two args which is the calling application name and the `flash` command from cargo.
-    let _ = args.next();
-    let _ = args.next();
+    args.remove(0);
+    args.remove(0);
+    // Remove possible `--chip <chip>` arguments as cargo build does not understand it.
+    if let Some(index) = args.iter().position(|x| *x == "--chip") {
+        args.remove(index);
+        args.remove(index);
+    }
```

Comment:
What's wrong with `skip()` and `take()`?

Reply:
I don't think it's very easy to use in this case. You would have to do some shenanigans which imo are way harder to understand :) I invite you to enlightn me tho, if you know a nicer solution :P